src/Compiler/Utilities/illib.fs

```diff
@@ -2,6 +2,8 @@
 
 namespace Internal.Utilities.Library
 
+#nowarn "842" //This attribute is not valid for use on this language element for `valueFactory` below
```

Comment:
This was fixed as well, here: https://github.com/dotnet/fsharp/pull/17093

We will need to remove once p4 releases.